### PR TITLE
LGA-3092 Implement Flask Skeleton Subtree

### PR DESCRIPTION
**Pull request by TawneeOwl**
## What does this pull request do?

Add the moj flask skeleton to the project as a subtree. This allows us to work on a fork of the project without the need for dependency on the project. It also allows us to update from the moj flask skeleton if required.

## Any other changes that would benefit highlighting?

Intentionally left blank.

## Checklist

- https://dsdmoj.atlassian.net/browse/LGA-3092

### CODE REVIEW

**Review by digitalali-moj — COMMENTED** [review]
Thanks, what a lovely idea! I want to learn more because it's an additional bit of complexity.

i wonder if this is a good time to add information on the readme file about how submodules work and what extra steps devs need to do to maintain correct working. Or maybe we can discuss this first.

For example how does it affect cloning the repo and what extra steps are needed.

How do we manage dependencies for the main project and the submodule and how does this then affect overall security?

How does the version controlling work for the submodule and does it affect version controlling of the API?

File: README.md [hunk 1]
```diff
@@ -4,3 +4,17 @@
...
+# MOJ Flask Skeleton
+
+The MOJ Flask Skeleton is set as a subtree from https://github.com/ministryofjustice/moj-flask-govuk-skeleton and is kept in the laa_access_civil_legal_aid folder.
+
+To retrieve the most recent updates from the flask skeleton, run:
+```shell
+git subtree pull --prefix=laa_access_civil_legal_aid https://github.com/ministryofjustice/moj-flask-govuk-skeleton.git main --squash
+```
+
+To contributing back to the MOJ Flask Skeleton Project with local updates:
```
Comment (digitalali-moj) [inline README.md:17]: Should we put (not recommended) type explanation here or alternative way of doing this?

File: README.md [hunk 2]
```diff
@@ -4,3 +4,17 @@
 
 
 A Flask based web service used by the public to get access to Civil Legal Aid services.
+
+# MOJ Flask Skeleton
+
+The MOJ Flask Skeleton is set as a subtree from https://github.com/ministryofjustice/moj-flask-govuk-skeleton and is kept in the laa_access_civil_legal_aid folder.
```
Comment (digitalali-moj) [inline README.md:10]: Is it worth adding a documentation link to "subtree" so people can easily find out what that means?

File: README.md [hunk 3]
```diff
@@ -4,3 +4,17 @@
...
+
+To retrieve the most recent updates from the flask skeleton, run:
```
Comment (digitalali-moj) [inline README.md:12]: Is it worth adding where this is done, e.g. inside feature branch?

**Review by digitalali-moj — APPROVED** [review]
Thanks for this and the run through, let's try this :)

File: README.md (hunk 1 above)
Comment (TawneeOwl) [reply README.md:17]: I've taken squash out of the main command and added it as an optional extra so it doesn't automatically overwrite

**Review by Pheonnexx — APPROVED** [review]

**Pull request merged**